time/range.go

```diff
@@ -61,6 +61,11 @@ func (r Range) Overlaps(other Range) bool {
 	return r.End.After(other.Start) && r.Start.Before(other.End)
 }
 
+// Duration returns the duration of the range.
+func (r Range) Duration() time.Duration {
+	return r.End.Sub(r.Start)
```

Comment:
The other tests all take as invariant that `Start` <= `End` so I didn't want to special case this one. Also, this would actually be fine since `time.Duration` is a type alias for `int64` and can be negative.